### PR TITLE
Clean up user_test.rb, consolidating tests for young and old students and other cleanup.

### PR DESCRIPTION
**Pull request by ashercodeorg**

### CODE REVIEW

File: dashboard/test/models/user_test.rb [hunk 1]
```diff
@@ -61,12 +61,6 @@ class UserTest < ActiveSupport::TestCase
     assert user.errors[:email].length == 1
   end
 
-  test "cannot create young user with invalid email" do
```
Comment (ashercodeorg) [inline dashboard/test/models/user_test.rb:64]: Duplicated by previous test as we no longer treat young students and old students differently with respect to email.

File: dashboard/test/models/user_test.rb [hunk 2]
```diff
@@ -351,28 +345,16 @@ class UserTest < ActiveSupport::TestCase
     assert user.secret_words !~ /SecretWord/ # using the actual word not the object to_s
   end
 
-  # TODO(asher): After the dust settles from PR#8582, consolidate the over and
-  # under 13 tests.
-  test 'students under 13 have hashed email not plaintext email' do
```
Comment (ashercodeorg) [inline dashboard/test/models/user_test.rb:356]: Combines an `under 13` and an `over 13` test into one test.

File: dashboard/test/models/user_test.rb [hunk 3]
```diff
@@ -351,28 +345,16 @@ class UserTest < ActiveSupport::TestCase
...
-
-  test 'students over 13 have hashed email not plaintext email' do
-    student = create :student, birthday: Date.new(1990, 10, 4), email: 'will_be_hashed@email.xx'
-
-    assert student.age.to_i > 13
     assert student.email.blank?
     assert student.hashed_email.present?
   end
 
   test 'teachers have hashed email and plaintext email' do
```
Comment (ashercodeorg) [inline dashboard/test/models/user_test.rb:355]: Birthday is now irrelevant for emails.

File: dashboard/test/models/user_test.rb [hunk 4]
```diff
@@ -491,9 +473,9 @@ class UserTest < ActiveSupport::TestCase
     assert ActionMailer::Base.deliveries.empty?
   end
 
-  test "send reset password for older student" do
```
Comment (ashercodeorg) [inline dashboard/test/models/user_test.rb:494]: Combines an `under 13` and an `over 13` test into one.

File: dashboard/test/models/user_test.rb [hunk 5]
```diff
@@ -491,9 +473,9 @@ class UserTest < ActiveSupport::TestCase
...
+  test 'send reset password for student' do
     email = 'email@email.xx'
-    student = create :student, age: 20, password: 'oldone', email: email
+    student = create :student, password: 'oldone', email: email
```
Comment (bcjordan) [inline dashboard/test/models/user_test.rb:478]: test password `'oldone'` isn't meaningful anymore, but also doesn't super matter :)

**Comment by bcjordan** [comment]
LGTM! Nice cleanup.

**Pull request merged**